aws/aws.go

```diff
@@ -251,5 +254,14 @@ func V2ConfigFromURLParams(ctx context.Context, q url.Values) (awsv2.Config, err
 		opts = append(opts, awsv2cfg.WithEndpointResolverWithOptions(customResolver))
 	}
 
+	// SDK v2 adds client-side rate limiting: https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/retries-timeouts/#client-side-rate-limiting
```

Comment:
Presumably AWS has some good reason for adding this, although they don't explain it in the doc you pointed to. So I'm not sure it's a great idea to disable it by default.

Can you make this a query parameter? (e.g., "disable_ratelimiting") and handle it similar to the others above (add it to the docstring too).

Reply:
> Presumably AWS has some good reason for adding this, although they don't explain it in the doc you pointed to. So I'm not sure it's a great idea to disable it by default.

I think these client side rate limits are causing more harm than good. The Terraform AWS provider disabled this with consultation with AWS: https://github.com/hashicorp/terraform-provider-aws/issues/36024#issuecomment-2008001009

Perhaps a better strategy is to leave this off by default and add the ability to configure the token bucket rate.

Reply:
How about we follow what Terraform did, and add a `rate_limiter_capacity` parameter. If the default is 0, then use `None` (I'm fine with changing that to be the default). If there's a value, then create a rate limiter with that capacity. Please link to https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/retries-timeouts/#client-side-rate-limiting in the documentation for the parameter.

Reply:
👍 Thanks, that's exactly what I think we should do.